lib/metasploit/framework/data_service/proxy/async_callback_data_proxy.rb

```diff
@@ -0,0 +1,24 @@
+module AsyncCallbackDataProxy
+
+  def async_callbacks(opts)
+    begin
+      self.data_service_operation do |data_service|
+        data_service.async_callbacks(opts)
+      end
+    rescue => e
+      self.log_error(e, "Problem retrieving async callback")
```

Comment:
As I mentioned with the URL, it might be beneficial to rename modules and update messages to include the term "payload" to help provide more context.